pkg/object/infos.go

```diff
@@ -4,15 +4,45 @@
 package object
 
 import (
+	"fmt"
+
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/cli-runtime/pkg/resource"
 	"sigs.k8s.io/kustomize/kyaml/kio/kioutil"
 )
 
+// InfosToObjMetas returns object metadata (ObjMetadata) for the
+// passed objects (infos); returns an error if one occurs.
+func InfosToObjMetas(infos []*resource.Info) ([]ObjMetadata, error) {
```

Comment:
Probably something for another PR, but we should see if we can reduce the number of times we end up converting between infos, unstructured, and ObjMetadata...

Reply:
We should probably keep objects in `Unstructured` format for most of the time, since `Info` and `ObjMetadata` can easily be created from this foundational format.

Reply:
...but you're right. That's probably best in another PR.

Reply:
Yeah, what gets a bit difficult is that converting from Unstructured to ObjMetadata without checks (like scope to know whether cluster-scoped or namespace-scoped) isn't necessarily safe. I'm working on a fix that would make this check immediately in the Applier.